### PR TITLE
Spec fixes

### PR DESCRIPTION
**Pull request by fastjames**
Before merging outstanding PRs, we would like to start with a green spec suite as a baseline.

### CODE REVIEW

**Comment by fastjames** [comment]
That's correct. The correct way to fix that is to backport the `factory_girl` -> `factory_bot` change to those versions of Solidus. I took a look at that last week but did not invest much time into it.

**Comment by fastjames** [comment]
This is stuck on Solidus 2.3 and 2.4, because those versions require `factory_girl` which is no longer available (replaced by `factory_bot`). This brings up the larger question of whether we need to backport that change, since those versions are still considered in maintenance.

File: app/models/solidus_subscriptions/subscription.rb [hunk 1]
```diff
@@ -7,7 +7,7 @@ class Subscription < ActiveRecord::Base
 
     PROCESSING_STATES = [:pending, :failed, :success]
 
-    belongs_to :user, class_name: Spree.user_class
+    belongs_to :user, class_name: Spree.user_class.to_s
```
Comment (jacobherrington) [inline app/models/solidus_subscriptions/subscription.rb]: Would `Spree.user_class.name` work here?

**Comment by jacobherrington** [comment]
I'm definitely in favor of merging this, however https://github.com/solidusio/solidus/pull/2835 is a blocker.

**Comment by kennyadsl** [comment]
Please rebase now, build should pass again, thanks!

**Comment by aitbw** [comment]
Superseeded by #61

**Pull request closed without merging**